kudoctl/src/subcommands/get/output.rs

```diff
@@ -0,0 +1,35 @@
+use anyhow::{Context, Result};
+use clap::ValueEnum;
+use serde::Serialize;
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ValueEnum, Debug)]
+pub enum OutputFormat {
+    /// Human readable format
+    HumanReadable,
+    /// JSON format
+    Json,
+    /// YAML format
+    Yaml,
+}
+
+/// Formats the output of a `get` command.
+pub fn format_output<T: Serialize + HumanReadable>(
+    output: T,
+    header: bool,
+    format: OutputFormat,
+) -> Result<String> {
+    match format {
+        OutputFormat::HumanReadable => output.human_readable(header, 80),
+        OutputFormat::Json => serde_json::to_string(&output).map_err(anyhow::Error::from),
+        OutputFormat::Yaml => serde_yaml::to_string(&output).map_err(anyhow::Error::from),
+    }
+    .context("Error formatting output")
+}
+
+/// Formats the output of a `get` command in the human readable format.
+pub trait HumanReadable {
```

Comment:
Can you please consider implementing the [Display](https://doc.rust-lang.org/std/fmt/trait.Display.html) trait for it instead?

Reply:
@sameo I considered the option, but i need to pass the `header` bool argument to it and I don’t know how to do it with the Display trait.

Reply:
Another thing that I wanted with this custom trait is the ability to limit the width of an output (with `max_col`).

Reply:
You could carry these options through e.g. the `GetWorkloadResponse` structure and implement `Display` on it.

Reply:
This would cause problems with the `Serialize` trait.

Reply:
Oh this should work, I’ll try that.

Reply:
Looks good. Can you please squash the 2 commits into a single one?

Reply:
I figured that this method won’t work anyway, in `get/mod.rs` the match statement returns multiple types (one `resources`, one for `resource`, one for `instance`...). I need to find a better way to handle printing the output.

Reply:
I tried a new way to handle my problem, what do you think ?